RadarSDK/RadarAPIClient.m

```diff
@@ -932,6 +932,50 @@ - (void)getMatrixFromOrigins:(NSArray<CLLocation *> *)origins
                     }];
 }
 
+- (void)sendEvent:(NSString *)name
+     withLocation:(CLLocation *_Nullable)location
+         metadata:(NSDictionary *_Nullable)metadata
+completionHandler:(RadarSendEventsAPICompletionHandler _Nonnull)completionHandler {
+    NSString *publishableKey = [RadarSettings publishableKey];
+    if (!publishableKey) {
+        return completionHandler(RadarStatusErrorPublishableKey, nil, nil);
+    }
+
+    NSMutableDictionary *params = [NSMutableDictionary new];
+    params[@"id"] = [RadarSettings _id];
+    params[@"installId"] = [RadarSettings installId];
+    params[@"userId"] = [RadarSettings userId];
+    params[@"deviceId"] = [RadarUtils deviceId];
+    params[@"name"] = name;
+    params[@"metadata"] = metadata;
+
+    NSString *host = [RadarSettings host];
+    NSString *url = [NSString stringWithFormat:@"%@/v1/events", host];
+    url = [url stringByAddingPercentEncodingWithAllowedCharacters:[NSCharacterSet URLQueryAllowedCharacterSet]];
+
+    NSDictionary *headers = [RadarAPIClient headersWithPublishableKey:publishableKey];
+
+    [self.apiHelper requestWithMethod:@"POST"
+                                  url:url
+                              headers:headers
+                               params:nil
+                                sleep:NO
+                           logPayload:YES
+                    completionHandler:^(RadarStatus status, NSDictionary *_Nullable res) {
+        if (status != RadarStatusSuccess || !res) {
+            return completionHandler(status, nil, nil);
+        }
+
+        id eventObj = res[@"event"];
+        RadarEvent *event = [RadarEvent eventFromObject:eventObj];
+        if (event) {
+            return completionHandler(RadarStatusSuccess, res, event);
+        }
+
+        completionHandler(RadarStatusErrorServer, nil, nil);
```

Comment:
Stick this in an `else`, so that it's clear.

Reply:
tbh I was just cribbing off the pattern for `trackWithLocation` - https://github.com/radarlabs/radar-sdk-ios/blob/develop/RadarSDK/RadarAPIClient.m#L283-L288

seems like we do it across all the fns in this file - https://github.com/radarlabs/radar-sdk-ios/blob/develop/RadarSDK/RadarAPIClient.m#L628-L631

I guess that might be why we have `return` in the first statement but not second